pkg/rollupsmachine/machine.go

```diff
@@ -0,0 +1,423 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package rollupsmachine
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/cartesi/rollups-node/pkg/emulator"
+)
+
+// Convenient type aliases.
+type (
+	Cycle  = uint64
+	Output = []byte
+	Report = []byte
+)
+
+type requestType uint8
+
+const (
+	DefaultInc = Cycle(10000000)
+	DefaultMax = Cycle(1000000000)
+
+	advanceStateRequest requestType = 0
+	inspectStateRequest requestType = 1
+
+	maxOutputs = 65536 // 2^16
+)
+
+// A RollupsMachine wraps an emulator.Machine and provides five basic functions:
+// Fork, Destroy, Hash, Advance and Inspect.
+type RollupsMachine struct {
+	// For each request, the machine will run in increments of Inc cycles,
+	// for no more than Max cycles.
+	//
+	// If these fields are left undefined,
+	// the machine will use the DefaultInc and DefaultMax values.
+	Inc, Max Cycle
+
+	address string
+
+	inner  *emulator.Machine
+	remote *emulator.RemoteMachineManager
+}
+
+// Load loads the machine stored at path into the remote server from address.
+// It then checks if the machine is in a valid state to receive advance and inspect requests.
+func Load(path, address string, config *emulator.MachineRuntimeConfig) (*RollupsMachine, error) {
+	// Creates the machine with default values for Inc and Max.
+	machine := &RollupsMachine{Inc: DefaultInc, Max: DefaultMax, address: address}
+
```

Comment:
Why are we using the default here? Shouldn't we check a config value and then use default only if it's empty?

Reply:
The interface for setting Inc and Max was confusing. I changed it and added documentation to make it clearer.